app/Libraries/Search/ForumSearchParams.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ *    Copyright 2015-2018 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Libraries\Search;
+
+use App\Libraries\Elasticsearch\SearchParams;
+
+class ForumSearchParams extends SearchParams
+{
+    // all public because lazy.
+
+    /** @var int|null */
+    public $forumId = null;
+
+    /** @var bool */
+    public $includeSubforums = false;
+
+    /** @var string|null */
+    public $queryString = null;
+
+    /** @var int|null */
+    public $topicId = null;
+
+    /** @var int|null */
+    public $username = null;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCacheKey() : string
```

Comment:
having `getCacheKey` function when `isCacheable` always returning `false` doesn't seem useful.

Reply:
The conditions for caching will be worked out later so it's all disabled except for the beatmap listing.

Reply:
and then the key will need updating thus rendering this one obsolete already :dancer: